pubpub/platform
# add toggle buttons to pub field form elements

## Issue(s) Resolved

Closes https://github.com/pubpub/pubpub/issues/3179

(Looks like this issue was accidentally created in the pubpub/pubpub repo)

## High-level Explanation of PR

This PR adds a button to the left side of form elements within the pub create form. Clicking the button toggles the disabled state for that field. By default all fields are enabled.

Disabled fields are omitted from the form when submitted. This means that submitting a form with all fields disabled will create a pub with no pub values.

## Test Plan

1. Use the create pub button to create a pub.
2. Create pubs with various configurations of disabled and enabled fields. Values of disabled fields should not be present on newly created pubs.
3. The form's submit button should only be disabled when _enabled_ buttons fail to validate. Invalid _disabled_ fields should not be considered.
4. Update a pub. The buttons should not be visible on the update form.

## Screenshots (if applicable)

<figure>
  <figcaption>The pub creation form with a disabled title field and invalid email field. The "Create Pub" button is disabled because of the enabled, invalid field.</figcaption>
  <p align="center">
    <img width="512" alt="Screenshot 2024-09-26 at 1 40 43 PM" src="https://github.com/user-attachments/assets/d7d8a0b8-8815-46b4-9e85-bb6903c4f680">
  </p>
</figure>

<figure>
  <figcaption>The same pub creation form after disabling the invalid email field. The "Create Pub" button is now enabled because the invalid field was disabled.</figcaption>
  <p align="center">
    <img width="513" alt="Screenshot 2024-09-26 at 1 40 49 PM" src="https://github.com/user-attachments/assets/c134b255-04c6-42ea-9ca9-d30cc17fa6d6">
  </p>
</figure>

<figure>
  <figcaption>The pub page of the newly created pub. Notice that the only field associated with the pub is "Member ID": the only enabled field on the form in the previous screenshot.
  <p align="center">
    <img width="661" alt="Screenshot 2024-09-26 at 1 41 12 PM" src="https://github.com/user-attachments/assets/f03ff0db-0fcd-43cb-879d-047a5acdcf9b">
  </p>
</figure>

## Notes
